swim/join_sender.go

```diff
@@ -65,6 +65,10 @@ type joinOpts struct {
 	size              int
 	maxJoinDuration   time.Duration
 	parallelismFactor int
+
+	// hostProvider is the HostProvider that this joinSender will use to
+	// enumerate bootstrap hosts.
+	hostProvider HostProvider
```

Comment:
Good point. In the magical utopia, we might not even be connecting via TCP, so it might not even be a host:port.